delta-app/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -290,18 +290,26 @@ private TableAssessmentResponse assessTable(SourceTable sourceTable, TableRegist
 
     // fetch detail about the table, then filter out columns that will not be read by the source
     TableDetail detail = tableRegistry.describeTable(db, table);
-    List<String> selectedPrimaryKeys = detail.getPrimaryKey().stream()
-      // if there are no columns specified, it means all columns (including primary keys) should be read
-      .filter(columnWhitelist.isEmpty() ? col -> true : columnWhitelist::contains)
-      .collect(Collectors.toList());
+    List<Problem> missingFeatures = new ArrayList<>(detail.getFeatures());
+    detail.getPrimaryKey().stream()
+      // if there are no columns specified, it means this primary key will be read/selected
+      .filter(columnWhitelist.isEmpty() ? col -> false : col -> !columnWhitelist.contains(col))
+      // add feature problem for any unselected primary key for table
+      .forEach(col -> missingFeatures.add(
```

Comment:
nit: I think it's better to have a single problem that mentions all of the columns, otherwise it could look kind of silly if it's missing 8 columns or something like that.

Reply:
Sure, since we do have primary key tag during table listing page. I agree to add just one problem here. I will do that change.